node_modules/oae-core/linkpreview/js/linkpreview.js

```diff
@@ -135,9 +137,19 @@ define(['jquery', 'oae.core', 'jquery.oembed'], function($, oae) {
                     embeddable = false;
                 }
             }
+            // Only embed text, html, images, and types that the browser has plugins to support
+            var type = widgetData.previews.targetType ? widgetData.previews.targetType.split(';')[0] : false;
+            var universalSupport = _.contains(['text/html', 'text/plain', 'image/png', 'image/jpeg', 'image/gif', 'image/svg+xml'], type);
+            var havePlugin = navigator.mimeTypes && navigator.mimeTypes[type] && navigator.mimeTypes[type].enabledPlugin;
+            var supported = widgetData.previews.targetType && (universalSupport || havePlugin);
+
             // The link can be embedded as an iFrame
-            if (embeddable) {
-                oae.api.util.template().render($('#linkpreview-iframe-template', $rootel), {'link': link}, $('#linkpreview-container'), $rootel);
+            if (embeddable && supported) {
+                if (type.substring(0, 6) === 'image/') {
+                    oae.api.util.template().render($('#linkpreview-image-template', $rootel), {'img': link, 'linkProfile': widgetData}, $('#linkpreview-container'), $rootel);
+                } else {
+                    oae.api.util.template().render($('#linkpreview-iframe-template', $rootel), {'link': link}, $('#linkpreview-container'), $rootel);
```

Comment:
When a link points to an external file (and therefore a screenshot can't be taken), we should be rendering the default link preview instead of the one where we try to display a screenshot of the page

![screen shot 2015-03-09 at 07 23 39](https://cloud.githubusercontent.com/assets/109850/6557045/43eae286-c62f-11e4-9668-fe2314d4b44f.png)